include/swift/AST/DiagnosticsSema.def

```diff
@@ -6312,9 +6312,8 @@ NOTE(objc_witness_objc_requirement,none,
 ERROR(no_opaque_return_type_of,none,
       "unable to resolve type for _opaqueReturnTypeOf attribute", ())
 
-
 ERROR(objc_observing_accessor, none,
-      "observing accessors are not allowed to be marked @objc", ())
+      "observers (%0) are not allowed to be marked '@objc'", (DescriptiveDeclKind))
```

Comment:
```suggestion
      "observers (%kindonly0) are not allowed to be marked '@objc'", (AccessorDecl *))
```

This way you can pass the accessor decl itself at the emission site, without computing its descriptive decl kind. The diagnostic formatter will do this for us.